### PR TITLE
kernel: bump 6.6 to 6.6.65

### PR DESCRIPTION
**Pull request by graysky2**
Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.6.65

Update to checksum only/no patches required a rebase. Per the changelog, this is a trivial upstream bump with only 4 commits, 2 of which being reverts.

Build system: x86/64
Build-tested: x86/64/AMD Cezanne
Run-tested: x86/64/AMD Cezanne

### CODE REVIEW

**Comment by graysky2** [comment]
I need someone to review `bcm27xx/patches-6.6/950-0829-vc4-hvs-Add-support-for-D0-register-changes.patch`

File: target/linux/generic/hack-6.6/780-usb-net-MeigLink_modem_support.patch [hunk 1]
```diff
@@ -10,22 +10,23 @@ Subject: [PATCH] net/usb/qmi_wwan: add MeigLink modem support
 
 --- a/drivers/net/usb/qmi_wwan.c
 +++ b/drivers/net/usb/qmi_wwan.c
-@@ -1084,12 +1084,18 @@ static const struct usb_device_id produc
+@@ -1084,6 +1084,11 @@ static const struct usb_device_id produc
  		USB_DEVICE_AND_INTERFACE_INFO(0x03f0, 0x581d, USB_CLASS_VENDOR_SPEC, 1, 7),
  		.driver_info = (unsigned long)&qmi_wwan_info,
  	},
-+	{	/* Meiglink SGM828 */
++	{       /* Meiglink SGM828 */
```
Comment (namiltd) [inline target/linux/generic/hack-6.6/780-usb-net-MeigLink_modem_support.patch:17]: The rest of the code is TAB

File: target/linux/generic/hack-6.6/780-usb-net-MeigLink_modem_support.patch [hunk 2]
```diff
@@ -10,22 +10,23 @@ Subject: [PATCH] net/usb/qmi_wwan: add MeigLink modem support
...
 +		USB_DEVICE_AND_INTERFACE_INFO(0x2dee, 0x4d49, USB_CLASS_VENDOR_SPEC, 0x10, 0x05),
 +		.driver_info = (unsigned long)&qmi_wwan_info,
 +	},
 +
+ 	{QMI_MATCH_FF_FF_FF(0x2c7c, 0x0122)},	/* Quectel RG650V */
  	{QMI_MATCH_FF_FF_FF(0x2c7c, 0x0125)},	/* Quectel EC25, EC20 R2.0  Mini PCIe */
  	{QMI_MATCH_FF_FF_FF(0x2c7c, 0x0306)},	/* Quectel EP06/EG06/EM06 */
- 	{QMI_MATCH_FF_FF_FF(0x2c7c, 0x0512)},	/* Quectel EG12/EM12 */
+@@ -1091,6 +1096,7 @@ static const struct usb_device_id produc
```
Comment (namiltd) [inline target/linux/generic/hack-6.6/780-usb-net-MeigLink_modem_support.patch:25]: The rest of the code is TAB

File: target/linux/generic/hack-6.6/780-usb-net-MeigLink_modem_support.patch [hunk 3]
```diff
@@ -10,22 +10,23 @@ Subject: [PATCH] net/usb/qmi_wwan: add MeigLink modem support
...
  	{QMI_MATCH_FF_FF_FF(0x2c7c, 0x0620)},	/* Quectel EM160R-GL */
  	{QMI_MATCH_FF_FF_FF(0x2c7c, 0x0800)},	/* Quectel RM500Q-GL */
  	{QMI_MATCH_FF_FF_FF(0x2c7c, 0x0801)},	/* Quectel RM520N */
-+	{QMI_MATCH_FF_FF_FF(0x05c6, 0xf601)},	/* MeigLink SLM750 */
++	{QMI_MATCH_FF_FF_FF(0x05c6, 0xf601)},   /* MeigLink SLM750 */
```
Comment (namiltd) [inline target/linux/generic/hack-6.6/780-usb-net-MeigLink_modem_support.patch:29]: The rest of the code is TAB

**Comment by Beginner-Go** [comment]
`bcm27xx/patches-6.6/950-0597-drm-vc4-hdmi-Avoid-hang-with-debug-registers-when-su.patch` Delete items that are not written to the update log

**Comment by namiltd** [comment]
In the file 950-0007-drm-vc4-Add-firmware-kms-mode.patch
replace
```
(vc4->is_vc5)
```
with
```
(vc4->vc4_gen == VC4_GEN_5)
```

File: target/linux/generic/hack-6.6/780-usb-net-MeigLink_modem_support.patch (hunk 1 above)
Comment (graysky2) [reply target/linux/generic/hack-6.6/780-usb-net-MeigLink_modem_support.patch]: fixed

File: target/linux/generic/hack-6.6/780-usb-net-MeigLink_modem_support.patch (hunk 3 above)
Comment (graysky2) [reply target/linux/generic/hack-6.6/780-usb-net-MeigLink_modem_support.patch]: fixed

**Comment by graysky2** [comment]
@namiltd - For the `:%s/(vc4->is_vc5)/(vc4->vc4_gen == VC4_GEN_5)/g` I assume all occurances?

File: target/linux/generic/hack-6.6/780-usb-net-MeigLink_modem_support.patch (hunk 2 above)
Comment (graysky2) [reply target/linux/generic/hack-6.6/780-usb-net-MeigLink_modem_support.patch:25]: fixed

File: target/linux/bcm27xx/patches-6.6/950-0007-drm-vc4-Add-firmware-kms-mode.patch [hunk 4]
```diff
@@ -2329,7 +2329,7 @@ Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
  
  	drm_atomic_helper_cleanup_planes(dev, state);
  
--	if (vc4->is_vc5) {
+-	if (vc4->vc4_gen == VC4_GEN_5) {
 +	if (vc4->is_vc5 && !vc4->firmware_kms) {
```
[the remark this answers is not in the document]
Comment (graysky2) [reply target/linux/bcm27xx/patches-6.6/950-0007-drm-vc4-Add-firmware-kms-mode.patch:2333]: Got it

**Comment by namiltd** [comment]
Revert to previous 950-0007-drm-vc4-Add-firmware-kms-mode.patch
and add to 950-0585-drm-vc4-Introduce-generation-number-enum.patch add a patch changing in file vc4_hvs.c:
```
@@ -224,6 +224,6 @@
 	if (!drm_dev_enter(drm, &idx))
 		return;
 
-	if (hvs->vc4->is_vc5)
+	if (hvs->vc4->vc4_gen == VC4_GEN_5)  
 		return;
 
 	/* The LUT memory is laid out with each HVS channel in order,
```

**Comment by graysky2** [comment]
@namiltd - sorry, I am not sure what you're suggesting.  Can you send a commit?

**Comment by namiltd** [comment]
[950-0007-drm-vc4-Add-firmware-kms-mode.patch](https://github.com/user-attachments/files/18068596/950-0007-drm-vc4-Add-firmware-kms-mode.patch)
I have corrected it with a text editor, so I apologize for any shortcomings.

**Comment by rmandrad** [comment]
tested pull from @namiltd successfully on the mediatek/filogic bananapi4

**Comment by namiltd** [comment]
Here you go: https://github.com/graysky2/openwrt/pull/4

**Comment by zxlhhyccc** [comment]
````
HDRINST usr/include/asm/poll.h
  INSTALL /__w/openwrt/openwrt/openwrt/build_dir/target-arm-openwrt-linux-muslgnueabi_musl/linux-bcm27xx_bcm2708/linux-6.6.64/user_headers/include
make[4]: Leaving directory '/__w/openwrt/openwrt/openwrt/build_dir/target-arm-openwrt-linux-muslgnueabi_musl/linux-bcm[27](https://github.com/openwrt/openwrt/actions/runs/12241818685/job/34147907735?pr=17217#step:31:28)xx_bcm2708/linux-6.6.64'
drivers/gpu/drm/vc4/vc4_hvs.c: In function 'vc4_hvs_lut_load':
drivers/gpu/drm/vc4/vc4_hvs.c:524:21: error: 'struct vc4_dev' has no member named 'is_vc5'
  524 |         if (hvs->vc4->is_vc5)
      |                     ^~
make[10]: *** [scripts/Makefile.build:243: drivers/gpu/drm/vc4/vc4_hvs.o] Error 1
make[9]: *** [scripts/Makefile.build:480: drivers/gpu/drm/vc4] Error 2
make[8]: *** [scripts/Makefile.build:480: drivers/gpu/drm] Error 2
make[7]: *** [scripts/Makefile.build:480: drivers/gpu] Error 2
make[6]: *** [scripts/Makefile.build:480: drivers] Error 2
make[6]: *** Waiting for unfinished jobs....
````

**Comment by namiltd** [comment]
@zxlhhyccc Fixed in https://github.com/graysky2/openwrt/pull/4.

**Comment by namiltd** [comment]
Download https://github.com/graysky2/openwrt/pull/4.patch
git reset --hard ff5f77ecdd218a9451bd6c531be86c6335fadd7a
git am location_of_file-4.patch
git push --force

**Comment by namiltd** [comment]
git commit --amend

and change:

Co-authored-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Signed-off-by: John Audia <therealgraysky@proton.me>

git push --force

**Comment by namiltd** [comment]
From:
```
Signed-off-by: John Audia <therealgraysky@proton.me>
Co-authoered-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
```
to
```
Co-authored-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Signed-off-by: John Audia <therealgraysky@proton.me>
``
And update the PR description (different than in Commit).

**Comment by namiltd** [comment]
Well, because I am as an author.
The easiest way:
- copy files somewhere
- rollback to before patch
- ecover files back
 - confirm and enter a comment

**Comment by namiltd** [comment]
.

**Comment by namiltd** [comment]
Build system: x86/64
Build-tested: ramips/tplink_archer-c5-v4
Run-tested: ramips/tplink_archer-c5-v4

**Comment by hauke** [comment]
You can change the author of a commit with `git commit --amend --author="John Audia <therealgraysky@proton.me>"`

**Comment by namiltd** [comment]
Errors are due tohttps://github.com/openwrt/openwrt/pull/16397#issuecomment-2538723118
You can make a quick patch.

**Comment by namiltd** [comment]
Please clean up

```
Signed-off-by: John Audia <therealgraysky@proton.me>
Co-authored-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
```
to
```
Co-authored-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Signed-off-by: John Audia <therealgraysky@proton.me>
```
so that this PR can be approved

**Review by namiltd — APPROVED** [review]

**Comment by robimarko** [comment]
Thanks! Rebased on top of main and merged!

**Pull request merged**